cdisc_rules_engine/services/define_xml_reader.py

```diff
@@ -273,6 +277,34 @@ def _get_variable_datatype(self, data_type):
         }
         return variable_type_map.get(data_type, data_type)
 
+    # TODO: subclass define.xml versions
+    # https://github.com/cdisc-org/cdisc-rules-engine/issues/390
+    def _get_origin_type(self, itemdef):
+        if not itemdef.Origin:
+            return
+        version_origin_map = {
+            DefineXMLVersions.DEFINE_XML_2_0_0: lambda origin: origin.Type,
+            DefineXMLVersions.DEFINE_XML_2_1_0: lambda origin: origin[0].Type,
+        }
+        return version_origin_map.get(DefineXMLVersions(self.get_define_version()))(
```

Comment:
I think the issue is here. Each time we call get_define_version() we are reading the file again. I think we should either save the version after reading it or find a quicker way  to get the version

Reply:
Pretty sure this is it. I thought about this, but forgot to revisit. Thanks for looking